File: 3477-fruits-into-baskets-ii/3477-fruits-into-baskets-ii.py
class Solution:
    def numOfUnplacedFruits(self, fruits: List[int], baskets: List[int]) -> int:
        n = len(fruits)  # 取得水果與籃子的數量（題目說 fruits 與 baskets 長度相同）
        used = [False] * n  # 用來紀錄哪些籃子已經被使用過
        unplaced = 0  # 統計無法放置的水果種類數量

        # 從左到右依序處理每一種水果
        for fruit in fruits:
            placed = False  # 標記該水果是否已成功放入籃子

            # 從左到右找第一個還沒被用掉，且容量大於等於水果數量的籃子
            for i in range(n):
                if not used[i] and baskets[i] >= fruit:
                    used[i] = True  # 標記籃子已使用
                    placed = True  # 標記水果已放置
                    break  # 找到合適籃子後就跳出迴圈

            # 如果整圈都找不到可用籃子，就累加未放置水果的數量
            if not placed:
                unplaced += 1

        return unplaced  # 回傳無法放置的水果種類數量

File: 3477-fruits-into-baskets-ii/3477-fruits-into-baskets-ii.cpp
class Solution {
public:
    int numOfUnplacedFruits(vector<int>& fruits, vector<int>& baskets) {
        int n = fruits.size();
        vector<bool> used(n, false); // 記錄每個籃子是否已被使用
        int unplaced = 0; // 統計無法放置的水果種類數量

        // 從左到右依序處理每一種水果
        for (int i = 0; i < n; ++i) {
            bool placed = false; // 記錄當前水果是否已成功放入籃子

            // 從左到右尋找第一個容量足夠且尚未使用的籃子
            for (int j = 0; j < n; ++j) {
                if (!used[j] && baskets[j] >= fruits[i]) {
                    used[j] = true; // 標記籃子已被使用
                    placed = true;  // 成功放置
                    break;          // 找到就跳出
                }
            }

            // 如果沒有任何籃子可以放這個水果
            if (!placed) {
                unplaced++;
            }
        }

        return unplaced;
    }
};